src/errors.rs

```diff
@@ -0,0 +1,25 @@
+pub type Result<T = ()> = std::result::Result<T, Error>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("{0}")]
+    FromUtf8(#[from] std::string::FromUtf8Error),
+    #[error("{0}")]
+    InvalidIp(#[from] std::net::AddrParseError),
```

Comment:
Use `InvalidNetworkAddress` instead of `InvalidIp` as `AddrParseError` covers more than just IP.